pipelines/smartseq2_single_sample/SS2SingleSample.wdl

```diff
@@ -0,0 +1,136 @@
+
+##### Runtime Attributes ######
+## ** This will change quite a bit i believe when the new spec of WDL comes out (WDL SPEC 3) but we should define all values we care about
+## ** even if they are the default values.  e.g. write `cpu: 1` instead of leaving it out because its the default.
+
+##### Comment Rules ######
+## **  Should comments have one or two `##` before a comment.  Maybe two for workflow or task descriptions and one for any other kind of comment?
+
+import "Hisat2QcPipeline.wdl" as RunHisatQCWDL  ## ** Should import alias follow some kind of pattern like ending in "WDL" or something else
+import "Hisat2RsemPipeline.wdl" as RunHisatRSEMWDL
+
+## ** Explanations above the workflow about what inputs it expects, what outputs/metrics it outputs
+## ** What, if any, assumptions are made in the pipeline
+## ** In general any kind of description you want to add above the workflow is nice
+## ** A good example of this can be found in these wdls
+## ** https://github.com/broadinstitute/dsde-pipelines/blob/develop/genomes_in_the_cloud/single_sample/PairedSingleSampleWf.wdl
+## ** https://github.com/broadinstitute/gatk/blob/master/scripts/mutect2_wdl/mutect2.wdl
+
+## ** Should comments / descriptions above a task definition be mandatory?
+
+## main pipeline:
+## QC track: HISAT2+Picard
+## this track will produce aligned bam file and a set of QC metrics
+## rsem quantification track: HISAT2+RSEM
+## this track involves hisat2 transcriptome alignment and RSEM gene expression estimation.
+workflow SmartSeq2SingleCell {
+  # load annotation
+  File gtf_file
+  File genome_ref_fasta
+  File rrna_intervals
+  File gene_ref_flat
+  # load index
+  File hisat2_ref_index
+  File hisat2_ref_trans_index
+  File rsem_ref_index
+  # ref index name
+  File hisat2_ref_name
+  File hisat2_ref_trans_name
+  # samples
+  String stranded
+  String sample_name
+  String output_name
+  File fastq1
+  File fastq2
+
+  call RunHisatQCWDL.RunHisat2Pipeline as QC {  ## ** Should callign a subworkflow be aliased some suffix or prefix so its easier to know when we're dealing with a workflow as opposed to a task?
+    input:
+      fastq_read1 = fastq1,
+      fastq_read2 = fastq2, 
+      gtf = gtf_file,
+      stranded = stranded,
+      ref_fasta = genome_ref_fasta,
+      rrna_interval = rrna_intervals,
+      ref_flat = gene_ref_flat,
+      hisat2_ref = hisat2_ref_index,
+      hisat2_ref_name = hisat2_ref_name,
+      sample_name = sample_name,
+      output_prefix = output_name + "_qc"
+  }
+
+  ## ** We should also figure out what we want the balance to be between moving stuff to subworkflows or getting rid of
+  ## ** them and moving that functionality up a level.  In this case do we like that QC and DATA are two different workflows
+  ## ** or would we rather move whats in those two wdls into here.  A good question to ask might be "Will we ever run
+  ## ** this subwokflow on its own?" If not, maybe it shouldnt be its own workflow?
```

Comment:
There are _also_ some stupid corner cases where we need subworkflows, e.g. optimus, due to limitations around nested scatters.

Reply:
yup so maybe the two cases we want to allow for `subworkflows` is getting around nested scatters and workflows that would be run on their own

Reply:
👍